src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateObject = {};
+  const sortDate = [];
+  const [, , , separatorFrom] = fromFormat;
+  const [, , , separatorTo] = toFormat;
+  const separateDate = date.split(separatorFrom);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY') {
+      if (separateDate[i] < 30) {
+        dateObject[fromFormat[i] + 'YY'] = '20' + separateDate[i];
+      } else {
+        dateObject[fromFormat[i] + 'YY'] = '19' + separateDate[i];
+      }
```

Comment:
The key generated for `dateObject` when `fromFormat[i]` is 'YY' is incorrect. The key should just be 'YY', not 'YY' + 'YY'. This will cause a mismatch when you try to access the values in the `dateObject` for the 'YY' key later on.